src/modules/cpu/host_fused_functions.hpp

```diff
@@ -4,12 +4,159 @@
+                         Rpp32f *batch_brightness, Rpp32f *batch_contrast,
+                         Rpp32f *batch_hue, Rpp32f *batch_saturation,
+                         Rpp32u outputFormatToggle, Rpp32u nbatchSize,
+                         RppiChnFormat chnFormat, Rpp32u channel)
+{
+    if(chnFormat == RPPI_CHN_PLANAR)
+    {
+        // NOT IMPLEMENTED
+    }
+    else if(chnFormat == RPPI_CHN_PACKED)
+    {
+        omp_set_dynamic(0);
+#pragma omp parallel for num_threads(nbatchSize)
+        for(int batchCount = 0; batchCount < nbatchSize; batchCount ++)
+        {
+            Rpp64u imageDimMax = batch_srcSizeMax[batchCount].height * batch_srcSizeMax[batchCount].width;
+
+            Rpp32f hue = batch_hue[batchCount];
+            Rpp32f saturation = batch_saturation[batchCount];
+            Rpp32f brightness = batch_brightness[batchCount];
+            Rpp32f contrast = batch_contrast[batchCount];
+
+            T *srcPtrImage, *dstPtrImage;
+            Rpp32u loc = 0;
+            compute_image_location_host(batch_srcSizeMax, batchCount, &loc, channel);
+            srcPtrImage = srcPtr + loc;
+            dstPtrImage = dstPtr + loc;
+
+            color_jitter_host(srcPtrImage, batch_srcSize[batchCount], batch_srcSizeMax[batchCount], dstPtrImage, brightness, contrast, hue, saturation, chnFormat, channel);
+
+            if (outputFormatToggle == 1)
+            {
+                T *dstPtrImageUnpadded = (T*) calloc(channel * batch_srcSize[batchCount].height * batch_srcSize[batchCount].width, sizeof(T));
+                T *dstPtrImageUnpaddedCopy = (T*) calloc(channel * batch_srcSize[batchCount].height * batch_srcSize[batchCount].width, sizeof(T));
+
+                compute_unpadded_from_padded_host(dstPtrImage, batch_srcSize[batchCount], batch_srcSizeMax[batchCount], dstPtrImageUnpadded, chnFormat, channel);
+
+                memcpy(dstPtrImageUnpaddedCopy, dstPtrImageUnpadded, channel * batch_srcSize[batchCount].height * batch_srcSize[batchCount].width * sizeof(T));
+
+                compute_packed_to_planar_host(dstPtrImageUnpaddedCopy, batch_srcSize[batchCount], dstPtrImageUnpadded, channel);
+
+                compute_padded_from_unpadded_host(dstPtrImageUnpadded, batch_srcSize[batchCount], batch_srcSizeMax[batchCount], dstPtrImage, RPPI_CHN_PLANAR, channel);
+
+                free(dstPtrImageUnpadded);
+                free(dstPtrImageUnpaddedCopy);
+            }
+        }
+    }
+
+    return RPP_SUCCESS;
+}
+
+template <typename T>
+RppStatus color_jitter_host(T* srcPtr, RppiSize srcSize, RppiSize srcSizeMax, T* dstPtr,
+                            Rpp32f brightness, Rpp32f contrast,
+                            Rpp32f hue, Rpp32f saturation,
+                            RppiChnFormat chnFormat, Rpp32u channel)
+{
+    if (chnFormat == RPPI_CHN_PLANAR)
+    {
+        // NOT IMPLEMENTED
+    }
+    else if (chnFormat == RPPI_CHN_PACKED)
+    {
+        Rpp32f hue_saturation_matrix[16] = {.299, .299, .299, 0.0,
```

Comment:
if you keep this matrix in aligned memory, you can use aligned loads and stores in SSE which will be faster

Reply:
Done. All 5 matrices changed to aligned_alloc() initialization.